### ISSUE TITLE
Not loading any configuration files until after one refresh period

### ISSUE THREAD

**Issue by thigger** [openhab/openhab#1898]
This will probably end up as wontfix but it's at least confusing if not a Proper Bug

OpenHab doesn't appear to make any attempt to load configuration files until after one scan period - eg with the following in openhab.cfg:

folder:items=180,items
folder:sitemaps=180,sitemap
folder:rules=200,rules
folder:scripts=180,script
folder:persistence=180,persist

There is a delay of 3 minutes before the items file is read after a restart (surprisingly common in our house!)

I'd suggest that on start, openhab should load all the configuration files in a sensible order (ie rules last) before starting its scanning behaviour. This might also be used to settle some of the other bugs on a similar topic eg #736 

Also as a result of this behaviour, if all scan times are set to be the same (I had them at 180), it's quite common for the rules used for initialisation of items to be executed before the items exist - hence the ugly '200' in my config file.

I've been using 1.4.0 for a while, just updated to 1.6.1 which seems to have the same behaviour.

**Comment by teichsta** [comment]
Hi @thigger,

thanks for sharing this. All files should be loaded directly on startup and are then revisited after the given time period. Could you please confirm that the files weren't loaded at startup directly?

Changing the startup behaviour is part of the core framework and should hence be discussed on the Eclipse SmartHome forum (https://www.eclipse.org/forums/index.php?t=thread&frm_id=271).

Thanks, Thomas E.-E.

**Comment by petrklus** [comment]
I am (and a few others) are seeing the same behaviour, see:

```
20:35:22.749 INFO  o.o.m.c.i.ModelRepositoryImpl[:79]- Loading model 'christmas.rules'
20:35:22.752 WARN  o.o.m.c.i.ModelRepositoryImpl[:58]- Configuration model 'christmas.rules' is either empty or cannot be parsed correctly!
20:35:22.755 INFO  o.o.m.c.i.ModelRepositoryImpl[:79]- Loading model 'bathroom.rules'
20:35:22.767 ERROR o.e.x.r.i.DefaultResourceDescription[:75]- christmas.rules (No such file or directory)
java.io.FileNotFoundException: christmas.rules (No such file or directory)
	at java.io.FileInputStream.open(Native Method)
	at java.io.FileInputStream.<init>(FileInputStream.java:146)
	at org.eclipse.emf.ecore.resource.impl.FileURIHandlerImpl.createInputStream(FileURIHandlerImpl.java:99)
	at org.eclipse.emf.ecore.resource.impl.ExtensibleURIConverterImpl.createInputStream(ExtensibleURIConverterImpl.java:354)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.load(ResourceImpl.java:1256)
	at org.eclipse.xtext.resource.impl.DefaultResourceDescription.computeExportedObjects(DefaultResourceDescription.java:73)
```

A "touch" of the same rule file later causes it to load correctly - just not during startup. Are there any plans on fixing it; is anybody looking into this? I understand that changes are planned for the future - but this a fundamental bug preventing unattended restart of the server.

**Comment by petrklus** [comment]
Moved up to test java8, and seeing the same issue. Sometimes I even need to "touch" the files multiple times, before they stop throwing the FileNotFound exception. Is there anything that could be done?

**Comment by petrklus** [comment]
Updated to the last version, still seeing the problem.

Is this only present on ARM platforms?

**Comment by steve-bate** [comment]
@petrklus This issue is discussing whether openhab attempts to load a file until the first refresh period. You seem to be having the problem that openhab attempts to load a file and fails. Your issue seems more related to #736. Do you agree? FWIW, I run on an ARM platform (RPi 3) using JSR223 rules and never see this problem.

**Comment by petrklus** [comment]
@steve-bate that is correct!

**Comment by kaikreuzer** [comment]
Closing as duplicate of #736 then.
